nixos/modules/services/security/crowdsec.nix

```
@@ -0,0 +1,945 @@
+                    share_context = false;
+                  };
+                  description = ''
+                    Attributes inside the console.yaml file.
+                  '';
+                };
+              };
+            };
+            description = ''
+              Console Configuration attributes
+            '';
+            default = { };
+          };
+        };
+      };
+    };
+  };
+  config =
+    let
+      cfg = config.services.crowdsec;
+      configFile = format.generate "crowdsec.yaml" cfg.settings.general;
+      simulationFile = format.generate "simulation.yaml" cfg.settings.simulation;
+      consoleFile = format.generate "console.yaml" cfg.settings.console.configuration;
+      patternsDir = pkgs.buildPackages.symlinkJoin {
+        name = "crowdsec-patterns";
+        paths = [
+          cfg.localConfig.patterns
+          "${lib.attrsets.getOutput "out" cfg.package}/share/crowdsec/config/patterns/"
+        ];
+      };
+
+      cscli = pkgs.writeShellScriptBin "cscli" ''
+        set -euo pipefail
+        # cscli needs crowdsec on it's path in order to be able to run `cscli explain`
+        export PATH="$PATH:${lib.makeBinPath [ cfg.package ]}"
+                sudo=exec
+        if [ "$USER" != "${cfg.user}" ]; then
+          ${
+            if config.security.sudo.enable then
+              "sudo='exec ${config.security.wrapperDir}/sudo -u ${cfg.user}"
+            else
+              ">&2 echo 'Aborting, cscli must be run as user `${cfg.user}`!'; exit 2"
+          }
+        fi
+        $sudo ${lib.getExe' cfg.package "cscli"} -c=${configFile} "$@"
+      '';
+
+      localScenariosMap = (map (format.generate "scenario.yaml") cfg.localConfig.scenarios);
+      localParsersS00RawMap = (
+        map (format.generate "parsers-s00-raw.yaml") cfg.localConfig.parsers.s00Raw
+      );
+      localParsersS01ParseMap = (
+        map (format.generate "parsers-s01-parse.yaml") cfg.localConfig.parsers.s01Parse
+      );
+      localParsersS02EnrichMap = (
+        map (format.generate "parsers-s02-enrich.yaml") cfg.localConfig.parsers.s02Enrich
+      );
+      localPostOverflowsS01WhitelistMap = (
+        map (format.generate "postoverflows-s01-whitelist.yaml") cfg.localConfig.postOverflows.s01Whitelist
+      );
+      localContextsMap = (map (format.generate "context.yaml") cfg.localConfig.contexts);
+      localNotificationsMap = (map (format.generate "notification.yaml") cfg.localConfig.notifications);
+      localProfilesFile = pkgs.writeText "local_profiles.yaml" ''
+        ---
+        ${lib.strings.concatMapStringsSep "\n---\n" builtins.toJSON cfg.localConfig.profiles}
+        ---
+      '';
+      localAcquisisionFile = pkgs.writeText "local_acquisisions.yaml" ''
+        ---
+        ${lib.strings.concatMapStringsSep "\n---\n" builtins.toJSON cfg.localConfig.acquisitions}
+        ---
+      '';
+
+      scriptArray = [
+        "set -euo pipefail"
+        "${lib.getExe cscli} hub update"
```

```suggestion
        "set -euo pipefail"
        "${lib.getExe' pkgs.coreutils "mkdir"} -p '${stateDir}/hub'"
        "${lib.getExe cscli} hub update"
```
I don’t know why this was an issue for me, considering I assume others have also tested this module, but I needed to add the above to avoid the following error.
```
systemd[1]: Starting CrowdSec agent...
crowdsec-setup[312691]: Downloading /var/lib/crowdsec/state/hub/.index.json
crowdsec-setup[312691]: Error: failed to update hub: failed to create temporary download file for /var/lib/crowdsec/state/hub/.index.json: open /var/lib/crowdsec/state/hub/.index.json.2211147961.download: no such file or directory
systemd[1]: crowdsec.service: Control process exited, code=exited, status=1/FAILURE
systemd[1]: crowdsec.service: Failed with result 'exit-code'.
systemd[1]: Failed to start CrowdSec agent.
```